compositor_render/src/event_loop.rs

```diff
@@ -0,0 +1,46 @@
+use std::sync::Arc;
+
+use compositor_chromium::cef;
+use signal_hook::{consts, iterator::Signals};
+
+pub struct EventLoop {
+    cef_ctx: Option<Arc<cef::Context>>,
+}
+
+impl EventLoop {
+    pub fn new(cef_ctx: Option<Arc<cef::Context>>) -> Self {
+        Self { cef_ctx }
+    }
+
+    /// Runs the event loop. It must run on the main thread.
+    /// Blocks the thread indefinitely
+    pub fn run(&self) -> Result<(), EventLoopRunError> {
+        match &self.cef_ctx {
+            Some(ctx) => self.cef_event_loop(ctx)?,
+            None => self.fallback_event_loop(),
+        }
+
+        Ok(())
+    }
+
+    fn cef_event_loop(&self, ctx: &cef::Context) -> Result<(), EventLoopRunError> {
+        if !ctx.currently_on_thread(cef::ThreadId::UI) {
+            return Err(EventLoopRunError::WrongThread);
+        }
+
+        ctx.run_message_loop();
+        Ok(())
+    }
+
+    /// Fallback event loop used when web renderer is disabled
+    fn fallback_event_loop(&self) {
+        let mut signals = Signals::new([consts::SIGINT]).unwrap();
```

Comment:
I missed that before, but neither the renderer nor the pipeline crate should depend on signal_hook. To be clear this is the correct place for the Event loop struct, but not for the code that handles interrupts.

Reply:
agreed